lib/std/priority_queue.zig

```diff
@@ -204,8 +204,13 @@ pub fn PriorityQueue(comptime T: type) type {
         pub fn update(self: *Self, elem: T, new_elem: T) !void {
             var update_index: usize = linearSearch(elem, self.items);
             assert (update_index >= 0 and update_index < self.items.len);
```

Comment:
Not finding `elem` is not an exceptional condition, return a proper error (eg. `return error.ElementNotFound`) and add a test for this.

Reply:
Yes, will do

Reply:
This assert is now useless (the index is unsigned so it's of course `>=0` and the upper bound is `<len` thanks to `indexOfScalar` and the error check above), you can safely remove it.